File: Baekjoon/수학/4948.py
# https://www.acmicpc.net/problem/4948

import sys
input = sys.stdin.readline

def prime_list(n):
    nums = [True] * (2 * n + 1)
    m = int((2 * n) ** 0.5)

    for i in range(2, m + 1):
        if nums[i] == True:
            for j in range(i + i, 2 * n + 1, i):
                nums[j] = False

    return [i for i in range(n + 1, 2 * n + 1) if nums[i] == True]

while True:
    n = int(input())
    if n == 0:
        break
    print(len(prime_list(n)))

File: Baekjoon/수학/4948.cpp
// https://www.acmicpc.net/problem/4948

#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
    while (true) {
        int n;
        cin >> n;
        if (n == 0) break;
        vector<bool> is_prime(2 * n + 1, true);
        is_prime[0] = is_prime[1] = false;
        for (int i = 2; i * i <= 2 * n; i++) {
            if (is_prime[i]) {
                for (int j = i * i; j <= 2 * n; j += i) {
                    is_prime[j] = false;
                }
            }
        }
        int cnt = 0;
        for (int i = n + 1; i <= 2 * n; i++) {
            if (is_prime[i]) cnt++;
        }
        cout << cnt << '\n';
    }
    return 0;
}